query-service-impl/src/test/java/org/hypertrace/core/query/service/prometheus/PrometheusRestClientTest.java

```diff
@@ -0,0 +1,144 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Map;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class PrometheusRestClientTest {
+
+  private static MockWebServer mockWebServer;
+  private OkHttpClient okHttpClient = new OkHttpClient();
+  private PrometheusRestClient prometheusRestClient = new PrometheusRestClient("localhost", 9090);
+
+  @BeforeAll
+  public static void setUp() throws IOException {
+    mockWebServer = new MockWebServer();
+    mockWebServer.start(9090);
+  }
+
+  @AfterAll
+  public static void tearDown() throws IOException {
+    mockWebServer.shutdown();
+  }
+
+  @Test
+  public void testSingleInstantQuery() throws IOException {
+    mockWebServer.enqueue(getSuccessMockResponse("promql_error_count_vector.json"));
+
+    PromQLInstantQuery query =
+        PromQLInstantQuery.builder()
+            .query("errorCount")
+            .evalTime(Instant.ofEpochMilli(1435781451000L))
+            .build();
+
+    Map<Request, PromQLMetricResponse> metricQueryResponses =
+        prometheusRestClient.executeInstantQuery(query);
+
+    Assertions.assertEquals(1, metricQueryResponses.size());
+  }
+
+  @Test
+  public void testSingleRangeQuery() throws IOException {
+    mockWebServer.enqueue(getSuccessMockResponse("promql_error_count_matrix.json"));
+
+    PromQLRangeQuery query =
+        PromQLRangeQuery.builder()
+            .query("errorCount")
+            .startTime(Instant.ofEpochMilli(1435781430000L))
+            .endTime(Instant.ofEpochMilli(1435781460000L))
```

Comment:
Can we have endtime = starttime + some value ?

Reply:
We can do that, but going with ease for corresponding API argument - https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries